tron/core/actionrun.py

```diff
@@ -737,6 +752,29 @@ def state_data(self):
             "trigger_timeout_timestamp": self.trigger_timeout_timestamp,
         }
 
+    @staticmethod
+    def from_json(state_data: Union[str, dict]):
```

Comment:
from what I recall we were always getting here a dictionary and I added the Union because I was getting this failure 
```
tron/core/actionrun.py:756:19: error: Argument 1 of "from_json" is incompatible with supertype "Persistable"; supertype defines the argument type as "str"  [override]
        def from_json(state_data: dict):
tron/core/actionrun.py:756:19: note: This violates the Liskov substitution principle
```
since it's arguments in the abstract class expect a str and not a dictionary. I'm not sure why we receive here a dictionary from the written json and if this was intended or not. This is why also in this function I dont do a json.loads.

Reply:
oh and I think cleanup run is being written as a string and that's why I left the union of both to handle both cases

Reply:
so in the to_json function: `` def to_json(state_data: dict) -> Optional[str]:`` 
state_data contains "runs" as a list of dictionary and then after the json.dumps it becomes a list of a string (this string is the dictionary)

Reply:
hmm I think I know what's happening, will try to fix it now